maven-core/src/main/java/org/apache/maven/project/MavenProject.java

```diff
@@ -1232,11 +1239,6 @@ private void deepCopy( MavenProject project )
             setDependencyArtifacts( Collections.unmodifiableSet( project.getDependencyArtifacts() ) );
         }
 
-        if ( project.getArtifacts() != null )
```

Comment:
This is now gone because of the deep copy?

Reply:
Yeah, please see this comment of mine: https://github.com/apache/maven/pull/527#discussion_r693556958